### ISSUE TITLE
"Run" button position

### ISSUE THREAD

**Issue by rotten77** [twibiral/obsidian-execute-code#208]
Please add position of the "Run" button into the settings. In case of short (one or two-line) snippet/code it overlays the "Copy" button:

![image](https://user-images.githubusercontent.com/2885380/212269849-dfce87e6-860e-471b-b0e4-bda46bb9f244.png)

**Comment by rotten77** [comment]
Hi, I tried options 1 and 2 by custom CSS just after I send this issue and I was not happy with the result :-)

I like the third option - the "play" icon but maybe place it at the left (before the code block).

My proposals are there. I think that the first one is most "easy to do", the second one looks nice but it can break the design. The third one is good because you can put the position in the settings (left top, left bottom, ...) but it looks different and may make users upset :-) On the other hand, the third option can be always displayed (not just by hovering).

<img width="791" alt="Screenshot 2023-01-21 at 7 41 54" src="https://user-images.githubusercontent.com/2885380/213847390-790dfa10-43f4-46d1-b8f6-5fb54faf7ad0.png">

**Comment by chlohal** [comment]
Thanks for your proposals!

I worry that the first one would hide the code. Maybe that's not as much of an issue if it's only visible on hover?

The second one would be a good match with the running indicator!

I think that the third option's positioning versatility is good, but I agree that users might dislike it because of the styling. We should also worry about accessibility -- to me, at least, they don't appear to be buttons, and that's very important to keep consistency across themes (i.e. we should be able to use the theme's "button" style)

**Comment by chlohal** [comment]
I just realized that we could also make an option to allow double-clicking on the box itself to run, rather than a visible button. What do you think?

**Comment by rotten77** [comment]
I thought about this too but I don't think it's a good idea. You can accidentally double-click and in that case, it would be unexpected behavior.

**Comment by hthuong09** [comment]
I would prefer something like rust by example site
<img width="833" alt="image" src="https://user-images.githubusercontent.com/8191266/214303263-f887fab2-57f5-43e1-84fa-c9c384dc52c3.png">

Can take a look here: https://doc.rust-lang.org/stable/rust-by-example/primitives/literals.html#literals-and-operators

**Comment by rotten77** [comment]
> I would prefer something like rust ...

I agree but this will change also the default behavior and design of Obsidian's copy button.

**Comment by rotten77** [comment]
I made a temporary solution with custom CSS for myself. It's ugly (I am too lazy to solve all CSS issues so I override them by "!important") but I can live with it :-)

![image](https://user-images.githubusercontent.com/2885380/214352683-11ac1bb4-6c49-40bb-b0d1-60fc98a09071.png)


```
button.run-code-button, button.copy-code-button, button.clear-button {
    border:1px solid #888;
    box-shadow: none !important;
    background: #333 !important;
    margin:0 !important;
    padding:0 !important;
    width:32px !important;
    height:24px !important;
    line-height: 24px;
    font-size:0.6em !important;
    text-align: center;
    position: absolute !important;

}
button.run-code-button, button.copy-code-button {
    top: 8px !important;
}
button.copy-code-button {right:16px !important;}
button.run-code-button {right:53px !important;}
button.clear-button {
    position: absolute !important;
    left: 16px !important;
    bottom: 8px !important;
}

button.run-code-button:hover, button.copy-code-button:hover, button.clear-button:hover {
    background:#555 !important;
}
```

**Comment by chlohal** [comment]
> I agree but this will change also the default behavior and design of Obsidian's copy button.

That's true-- we can always make it available as an option, but maybe not a recommended one?

**Comment by rotten77** [comment]
I discovered that some other extensions also change the default copy button style so it can be messy in general :-)

**Comment by luckman212** [comment]
Is there a way to try out the "play" button icon (instead of Run button) right now using a fork or editing the CSS? I have the same problem of lots of little 1-liners in my notes that I try to copy but the "Run" button from this extension is overlapping/blocking it... 🙏

**Comment by rotten77** [comment]
Hi @luckman212, I update my CSS snippet for replacing the "Run" text with "play" icon:

```
button.run-code-button, button.copy-code-button, button.clear-button {
    border:1px solid #888;
    box-shadow: none !important;
    background: #333 !important;
    margin:0 !important;
    padding:0 !important;
    width:32px !important;
    height:24px !important;
    line-height: 24px;
    font-size:0.6em !important;
    text-align: center;
    position: absolute !important;

}
button.run-code-button, button.copy-code-button {
    top: 8px !important;
}
button.copy-code-button {right:16px !important;}
button.run-code-button {right:53px !important;}
button.clear-button {
    position: absolute !important;
    left: 16px !important;
    bottom: 8px !important;
}

button.run-code-button:hover, button.copy-code-button:hover, button.clear-button:hover {
    background:#555 !important;
}
/* Play icon */
button.run-code-button {
    width:32px;
    overflow:hidden;
    font-size: 0 !important;
}
button.run-code-button::before {
    content: '\25B6';
    font-size: 16px;
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%); /* Center the icon */
}
```

This is how it looks:

![image](https://github.com/twibiral/obsidian-execute-code/assets/2885380/3e0378f2-fcf2-4247-9dd5-7b5aff49b8d5)

How to "install":

At the bottom of `Settings > Appearance` you can find a "CSS snippets" section - click on the folder button (open snippets folder) on the right side and place the CSS file with this content in the folder that was opened. You can change CSS properties to match your theme/preferences (color, size, ...). You can ask ChatGPT to change the CSS as well :-)

**Comment by luckman212** [comment]
@rotten77 This is awesome! 🙏 

I tweaked it a bit to arrive at this

<img width="363" alt="484acbf4016308f316f08e4060482290ca4ff7b2_gh" src="https://github.com/twibiral/obsidian-execute-code/assets/1992842/2c26be1a-8521-4a64-a94f-0f5df0ad3bb2">
